labonneboite/common/maps/README.md

```diff
@@ -0,0 +1,360 @@
+
+
+#### 2/ Use this data with Elastic Search
+
+Elastic Search understands pretty well geometric data, including polygons. We just have to pass it to the Elastic Search query. This is made in two steps:
+
+`labonneboite/web/search/views.py`
+
+```
+@searchBlueprint.route('/entreprises')
+def entreprises():
+    # Convert request arguments to fetcher parameters
+    parameters = get_parameters(request.args)
+
+    # Fetch offices and alternatives
+    # Other filters are hidden in this documentation for readability purposes.
+    fetcher = search_util.HiddenMarketFetcher(
+        location,
+        travel_mode=parameters['travel_mode'],
+        duration=parameters['duration'],
+        # ...
+    )
+```
+
+
+`labonneboite/common/search.py`
+
+```
+def build_json_body_elastic_search(...):
+    # ...
+    # Add a filter in Elastic Search
+    if duration is not None:
+        # Retrieve cached data
+        isochrone = travel.isochrone((latitude, longitude), duration, mode=travel_mode)
+        if isochrone:
+            for polygon in isochrone:
+                should_filters.append({
+                    "geo_polygon": {
+                        "locations": {
+                            "points": [[p[1], p[0]] for p in polygon]
+                        }
+                    }
+                })
+```
+
+
+#### 3/ Display filtered results to the user
+
+Once results are ready in the `search.entreprises` view, two templates can be shown:
+- if request is an AJAX call: `search/results_content.html`
+- if not: `search/results.html`
+
+
+`labonneboite/web/search/views.py`
+
+```
+@searchBlueprint.route('/entreprises')
+def entreprises():
+    # Render different template if it's an ajax call
+    template = 'search/results.html' if not request.is_xhr else 'search/results_content.html'
+```
+
+Both are located here: `labonneboite/templates/`
+
+
+#### 4/ Show commute time for each enterprise
+
+![](readme_images/commuting_time.png)
+
+Commute time is not part of former templates, instead it's generated by this script: `labonneboite/web/static/js/results.js`.
+
+```
+(function ($) {
+
+  $(document).on('lbbready', function () {
+    // ...
+    updateTravelDurations();
+  }
+}
+```
+
+:information_source: Durations are retrieved 5 by 5 to avoid timeouts from upstream servers.
+
+
+## Cache
+
+Retrieving results from third-party APIS is expensive and increases the risk of errors. That's why we use a cache system to store data for a limited time.
+
+
+### Available caches
+
+Three caches are available, depending on your needs and on your configuration:
+- `LocalCache`: stores data in memory as a dictionary. :warning: This is highly inefficient and can lead to data loss, so it should not be used in production.
+- `DummyCache`: returns None, same as saying "I don't have this data in my cache". Not used for the moment.
```

Comment:
This is used only in tests, unless I'm mistaken.

Reply:
Thanks! I had misread the settings. Correcting it.

Reply:
Hum...
I'm not sure about this DummyCache. In test settings, "dummy" is the vendors backend, not local cache one [code](https://github.com/StartupsPoleEmploi/labonneboite/blob/master/labonneboite/conf/common/overrides/test.py#L28).
This defines which file to import : [code](https://github.com/StartupsPoleEmploi/labonneboite/blob/master/labonneboite/common/maps/vendors/__init__.py#L4).
[`vendors/dummy.py`](https://github.com/StartupsPoleEmploi/labonneboite/blob/master/labonneboite/common/maps/vendors/dummy.py) does not use DummyCache.
And TRAVEL_CACHE is not overwritten in test settings.

Reply:
Petit up @regisb sur cette question. Merci !